src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -19,7 +23,33 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        TreeSet<Character> charSet = new TreeSet<>();
+        StringBuilder charBuilder = new StringBuilder();
+        String resultString = "";
+        File file = new File(fileName);
+        if (!file.exists()) {
+            throw new FileNotFoundException("File doesn't exist.");
+        }
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+            while (fileInputStream.available() > 0) {
+                char alphabetChar = (char) fileInputStream.read();
+                if (Character.isLetter(alphabetChar)) {
+                    charSet.add(Character.toLowerCase(alphabetChar));
+                }
+            }
+            for (Character nextCharacter : charSet) {
+                charBuilder.append(nextCharacter);
```

Comment:
stop append after 5 chars is appended. As result you will not need logic below of substing. Also you will be able to remove resultString var.

Reply:
done